Astropy 3.2.2 test failures with Numpy 1.14.6

Some of the tests for Astropy 3.2.2 are failing with Numpy 1.14.6 which is causing the wheels to not be built for Python 3.7 (see https://github.com/astropy/astropy/issues/9371 for example). To reproduce with Python 3.7:

```
python -m venv python/debug-wheel
source ~/python/debug-wheel/bin/activate
pip install numpy==1.14.6
pip install astropy --no-binary :all:
pip install pytest-astropy
python -c "import astropy;astropy.test(args='-k test_regression_futuretimes_4302')"```
```

The error is then:

```
__________________________________________________________ test_regression_futuretimes_4302 ___________________________________________________________

    def test_regression_futuretimes_4302():
        """
        Checks that an error is not raised for future times not covered by IERS
        tables (at least in a simple transform like CIRS->ITRS that simply requires
        the UTC<->UT1 conversion).
    
        Relevant comment: https://github.com/astropy/astropy/pull/4302#discussion_r44836531
        """
        from astropy.utils.exceptions import AstropyWarning
    
        # this is an ugly hack to get the warning to show up even if it has already
        # appeared
        from astropy.coordinates.builtin_frames import utils
        if hasattr(utils, '__warningregistry__'):
            utils.__warningregistry__.clear()
    
        with catch_warnings() as found_warnings:
            future_time = Time('2511-5-1')
            c = CIRS(1*u.deg, 2*u.deg, obstime=future_time)
>           c.transform_to(ITRS(obstime=future_time))

python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/tests/test_regression.py:226: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/baseframe.py:1202: in transform_to
    return trans(self, new_frame)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/transformations.py:1377: in __call__
    curr_coord = t(curr_coord, curr_toframe)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/transformations.py:980: in __call__
    return supcall(fromcoord, toframe)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py:79: in cirs_to_itrs
    pmat = cirs_to_itrs_mat(itrs_frame.obstime)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py:31: in cirs_to_itrs_mat
    xp, yp = get_polar_motion(time)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/utils.py:41: in get_polar_motion
    xp, yp, status = iers.IERS_Auto.open().pm_xy(time, return_status=True)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:660: in open
    cls.iers_table = cls.read(file=filename)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:507: in read
    table = cls._combine_a_b_columns(iers_a)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:469: in _combine_a_b_columns
    is_predictive = (table['UT1Flag_A'] == 'P') | (table['PolPMFlag_A'] == 'P')
python/debug-wheel/lib/python3.7/site-packages/astropy/table/column.py:986: in _compare
    return result.data if isinstance(result, Column) else result
python/debug-wheel/lib/python3.7/site-packages/astropy/table/column.py:1278: in data
    out.fill_value = self.fill_value
python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:3375: in __setattr__
    super(MaskedArray, self).__setattr__(attr, value)
python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:3659: in set_fill_value
    target = _check_fill_value(value, self.dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

fill_value = 'N', ndtype = dtype('bool')

    def _check_fill_value(fill_value, ndtype):
        """
        Private function validating the given `fill_value` for the given dtype.
    
        If fill_value is None, it is set to the default corresponding to the dtype.
    
        If fill_value is not None, its value is forced to the given dtype.
    
        The result is always a 0d array.
        """
        ndtype = np.dtype(ndtype)
        fields = ndtype.fields
        if fill_value is None:
            fill_value = default_fill_value(ndtype)
        elif fields:
            fdtype = [(_[0], _[1]) for _ in ndtype.descr]
            if isinstance(fill_value, (ndarray, np.void)):
                try:
                    fill_value = np.array(fill_value, copy=False, dtype=fdtype)
                except ValueError:
                    err_msg = "Unable to transform %s to dtype %s"
                    raise ValueError(err_msg % (fill_value, fdtype))
            else:
                fill_value = np.asarray(fill_value, dtype=object)
                fill_value = np.array(_recursive_set_fill_value(fill_value, ndtype),
                                      dtype=ndtype)
        else:
            if isinstance(fill_value, basestring) and (ndtype.char not in 'OSVU'):
                err_msg = "Cannot set fill value of string with array of dtype %s"
>               raise TypeError(err_msg % ndtype)
E               TypeError: Cannot set fill value of string with array of dtype bool

python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:473: TypeError
```

I have bisected this and it looks like this comes from https://github.com/astropy/astropy/pull/8904.

@mhvk, since this was your PR, is there any chance you could look into this? I know we aren't supporting Numpy 1.14 anymore in master, but it would be good to get all tests passing and do a last 3.2.x release before 4.0.

--- a/astropy/table/column.py
+++ b/astropy/table/column.py
@@ -23,6 +23,7 @@ from astropy.utils.console import color_print
 from astropy.utils.metadata import MetaData
 from astropy.utils.data_info import BaseColumnInfo, dtype_info_name
 from astropy.utils.misc import dtype_bytes_or_chars
+from astropy.utils.compat import NUMPY_LT_1_15
 from . import groups
 from . import pprint
 from .np_utils import fix_column_name
@@ -979,6 +980,11 @@ class Column(BaseColumn):
 
             if self.dtype.char == 'S':
                 other = self._encode_str(other)
+
+            if NUMPY_LT_1_15 and (isinstance(self, np.ma.MaskedArray) or
+                                  isinstance(other, np.ma.MaskedArray)):
+                # On numpy 1.14, MaskedArray subclass comparison is broken.
+                return getattr(self.data, op)(other)
 
             # Now just let the regular ndarray.__eq__, etc., take over.
             result = getattr(super(), op)(other)

--- a/astropy/utils/compat/numpycompat.py
+++ b/astropy/utils/compat/numpycompat.py
@@ -7,7 +7,7 @@ from astropy.utils import minversion
 
 
 __all__ = ['NUMPY_LT_1_14', 'NUMPY_LT_1_14_1', 'NUMPY_LT_1_14_2',
-           'NUMPY_LT_1_16', 'NUMPY_LT_1_17']
+           'NUMPY_LT_1_15', 'NUMPY_LT_1_16', 'NUMPY_LT_1_17']
 
 # TODO: It might also be nice to have aliases to these named for specific
 # features/bugs we're checking for (ex:
@@ -15,5 +15,6 @@ __all__ = ['NUMPY_LT_1_14', 'NUMPY_LT_1_14_1', 'NUMPY_LT_1_14_2',
 NUMPY_LT_1_14 = not minversion('numpy', '1.14')
 NUMPY_LT_1_14_1 = not minversion('numpy', '1.14.1')
 NUMPY_LT_1_14_2 = not minversion('numpy', '1.14.2')
+NUMPY_LT_1_15 = not minversion('numpy', '1.15')
 NUMPY_LT_1_16 = not minversion('numpy', '1.16')
 NUMPY_LT_1_17 = not minversion('numpy', '1.17')

